packages/cactus-plugin-ledger-connector-besu/src/main/typescript/web-services/watch-events-v1-endpoint.ts

```diff
@@ -0,0 +1,149 @@
+import type { Socket as SocketIoSocket } from "socket.io";
+import { stringify } from "safe-stable-stringify";
+import {
+  Abi,
+  Transport,
+  PublicClient as ViemPublicClient,
+  WatchContractEventParameters,
+} from "viem";
+
+import {
+  Logger,
+  Checks,
+  bigIntToDecimalStringReplacer,
+} from "@hyperledger/cactus-common";
+import { LogLevelDesc, LoggerProvider } from "@hyperledger/cactus-common";
+import {
+  ViemV2242WatchEventsV1Progress,
+  WatchEventsV1Request,
+} from "../generated/openapi/typescript-axios";
+import { WatchEventsV1 } from "../generated/openapi/typescript-axios";
+import { watchEventsV1Impl } from "../impl/watch-events-v1/watch-events-v1-impl";
+import { ensure0xPrefix } from "../common/ensure-0x-prefix";
+
+export interface IWatchEventsV1EndpointOptions {
+  readonly logLevel?: LogLevelDesc;
+  readonly socket: SocketIoSocket;
+  readonly viemClient: ViemPublicClient;
+}
+
+export class WatchEventsV1Endpoint {
```

Comment:
@petermetz, I would just suggest renaming classes/types to something like SolidityEvent or ContractEvent, just like the PR title. This may be a bit misleading since there is usually the possibility of listening for multiple events beyond the ones emitted by contracts (e.g., pending Transactions, new Heads, etc.)